packages/cloud_functions/cloud_functions/ios/cloud_functions/Sources/cloud_functions/FirebaseFunctionsPlugin.swift

```diff
@@ -0,0 +1,179 @@
+
+  public func didReinitializeFirebaseCore(_ completion: @escaping () -> Void) {
+    completion()
+  }
+
+  public func pluginConstants(for firebaseApp: FirebaseApp) -> [AnyHashable: Any] {
+    [:]
+  }
+
+  @objc public func firebaseLibraryName() -> String {
+    "flutter-fire-fn"
+  }
+
+  @objc public func flutterChannelName() -> String {
+    kFLTFirebaseFunctionsChannelName
+  }
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let binaryMessenger: FlutterBinaryMessenger
+
+    #if os(macOS)
+      binaryMessenger = registrar.messenger
+    #elseif os(iOS)
+      binaryMessenger = registrar.messenger()
+    #endif
+
+    let channel = FlutterMethodChannel(
+      name: kFLTFirebaseFunctionsChannelName,
+      binaryMessenger: binaryMessenger
+    )
+    let instance = FirebaseFunctionsPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    guard call.method == "FirebaseFunctions#call" else {
+      result(FlutterMethodNotImplemented)
+      return
+    }
+
+    guard let arguments = call.arguments as? [String: Any] else {
+      result(FlutterError(code: "invalid_arguments",
+                          message: "Invalid arguments",
+                          details: nil))
+      return
+    }
+
+    httpsFunctionCall(arguments: arguments) { success, error in
+      if let error {
+        result(error)
+      } else {
+        result(success)
+      }
+    }
+  }
+
+  private func httpsFunctionCall(arguments: [String: Any],
+                                 completion: @escaping (Any?, FlutterError?) -> Void) {
+    let appName = arguments["appName"] as? String ?? ""
+    let functionName = arguments["functionName"] as? String
+    let functionUri = arguments["functionUri"] as? String
+    let origin = arguments["origin"] as? String
+    let region = arguments["region"] as? String
+    let timeout = arguments["timeout"] as? Double
+    let parameters = arguments["parameters"]
+    let limitedUseAppCheckToken = arguments["limitedUseAppCheckToken"] as? Bool ?? false
+
+    let app = FLTFirebasePlugin.firebaseAppNamed(appName)!
+
+    let functions = Functions.functions(app: app, region: region ?? "")
+
+    if let origin, !origin.isEmpty,
+       let url = URL(string: origin),
+       let host = url.host,
+       let port = url.port {
+      functions.useEmulator(withHost: host, port: port)
+    }
+
+    let options = HTTPSCallableOptions(requireLimitedUseAppCheckTokens: limitedUseAppCheckToken)
+
+    let function: HTTPSCallable
+
+    if let functionName, !functionName.isEmpty {
+      function = functions.httpsCallable(functionName, options: options)
+    } else if let functionUri, !functionUri.isEmpty,
+              let url = URL(string: functionUri) {
+      function = functions.httpsCallable(url, options: options)
+    } else {
+      completion(nil, FlutterError(
+        code: "unknown",
```

Comment:
```suggestion
        code: "IllegalArgumentException",
```
Should match the previous implementation